Price range and other bugs

darekrossman opened the pull request:
- handle price range and sorting display
  - Product tile prices with range now say "From 49.99"
- handle PDP price display
  - Range price is display "min - max" for range price products
  - when a variant is matched from selections, the variant's actual price displays instead
- ensure default sort value is visible in UI even when sort param is absent
- remove auto overflow on PDP image thumbnails, was causing awkward vertical scroll unnecessarily 
- added phone field to address forms, auto formats for US/CA but not for UK (keeping simple)

darekrossman committed:
implement sfcc checkout

darekrossman committed:
Refactor SFCC SDK implementation and configuration

darekrossman committed:
Refactor product recommendation logic and remove unused OCAPI module

darekrossman committed:
Remove default values from payment form fields

darekrossman committed:
ghost commit

darekrossman committed:
Add environment variable registration in instrumentation.ts

darekrossman committed:
Add environment variable registration function to instrumentation

darekrossman committed:
node runtime check

darekrossman committed:
update and use getCollectionProducts for cached category search

darekrossman committed:
Remove TODO comment regarding product attribute name handling in reshapeProduct function

darekrossman committed:
Merge branch 'main' into sdk-replacement and some cleaning

darekrossman committed:
Merge branch 'sdk-replacement' into sfcc-checkout

darekrossman committed:
ts fix

darekrossman committed:
minor design feedback

darekrossman committed:
Add collapsible summary component and update dependencies

- Introduced `CollapsibleSummary` component for order summary in the checkout layout.
- Added `@radix-ui/react-collapsible` dependency to enable collapsible functionality.
- Adjusted spacing in various forms for consistent layout.
- Updated `package.json` and `pnpm-lock.yaml` to reflect new dependency.

darekrossman committed:
remove explicit white text color to hanbdle light and dark mode

darekrossman committed:
Merge branch 'sfcc-checkout' into checkout-summary-enhancements

darekrossman committed:
clean up styling for light mode

darekrossman committed:
fix runtime check

darekrossman committed:
Merge branch 'sfcc-checkout' into checkout-summary-enhancements

darekrossman committed:
Refactor checkout forms to use AddressForm component and update billing address handling

- Replaced inline address fields in InformationForm and PaymentForm with AddressForm component for better code organization and reusability.
- Updated the updateBillingAddress function to utilize the new billingAddressSchema.
- Introduced billing address handling logic in PaymentForm to conditionally display the AddressForm based on user input.
- Added prefixing functionality to address schemas for improved data structure.

darekrossman committed:
Merge branch 'main' into sfcc-checkout

darekrossman committed:
Merge branch 'sfcc-checkout' into checkout-summary-enhancements

darekrossman committed:
Merge branch 'main' into checkout-summary-enhancements

darekrossman committed:
Merge branch 'main' into billing-form

darekrossman committed:
Enhance product variant selection and URL management

- Added cursor pointer style for buttons and roles.
- Updated product context to allow URL updates with replace option.
- Improved variant selector to handle color option transitions and state updates.
- Cleaned up code formatting for consistency.

darekrossman committed:
Enhance product and variant handling in checkout components

- Added display of selected options in the checkout cart for better user clarity.
- Updated variant selector to handle option values as objects, improving state management and availability checks.
- Refactored product reshaping logic to include option values with IDs and names for consistency across components.
- Introduced variationValues in product types for better data handling.

darekrossman committed:
Enhance payment form validation and accessibility

- Added regex patterns for cardholder name and security code to enforce input validation.
- Updated error message for security code to include an ID and role for improved accessibility.
- Ensured cardholder name only accepts letters and spaces, while security code requires 3-4 digits.

darekrossman committed:
Enhance checkout status component with billing address display

- Added billing address handling to the CheckoutStatus component.
- Implemented logic to check if the billing address is the same as the shipping address.
- Updated the UI to conditionally display the billing address or a message indicating it matches the shipping address.

darekrossman committed:
Refactor and clean up component imports and formatting

- Standardized import statements across components to use consistent quotation marks.
- Improved code formatting for better readability and consistency in the CartModal, Input, and Select components.
- Enhanced conditional rendering in the CartModal for shipping amount display.
- Minor adjustments to ensure uniformity in class names and props handling.

darekrossman committed:
Merge branch 'checkout-summary-enhancements' into billing-form

darekrossman committed:
Remove placeholder text from input fields in checkout forms for a cleaner UI experience

darekrossman committed:
Refactor CollapsibleSummary component and adjust layout in Checkout

- Moved CollapsibleSummary component to be conditionally rendered within the main layout for better visibility on mobile devices.
- Updated styling in CollapsibleSummary for improved aesthetics and consistency, including adjustments to padding and border properties.

darekrossman committed:
Implement PostalCodeInput component and refactor credit card utilities

- Introduced a new PostalCodeInput component to handle postal code formatting for US, CA, and UK.
- Updated the CreditCardInput component to use the new form-helpers utility for credit card formatting.
- Refactored existing imports to align with the new form-helpers module for consistency across components.

darekrossman committed:
Remove console log from CheckoutConfirmationPage

darekrossman committed:
debugging search

darekrossman committed:
Add debugging logs to Search component for query tracking

darekrossman committed:
Add submit button to Search component for improved functionality

darekrossman committed:
Update Search component to disable prefetching for improved performance

darekrossman committed:
rollback, turn off prefetch

darekrossman committed:
Refactor product pricing and display components for improved clarity and functionality

- Updated ProductPage to utilize new ProductPrice component for better price handling.
- Refactored Label and Price components to support minimum and maximum price ranges.
- Adjusted GridTileImage and ProductGridItems to accommodate new pricing structure.
- Enhanced reshapeProduct function to streamline price data handling from the API.
- Cleaned up imports and standardized quotation marks across files for consistency.

darekrossman committed:
update prop usage

darekrossman committed:
Refactor dropdown and filter item components for consistency and functionality

darekrossman committed:
remove auto overflow on image thumbs

darekrossman committed:
Add PhoneInput component and integrate into AddressForm

- Introduced PhoneInput component for handling phone number input with country-specific formatting.
- Updated AddressForm to include PhoneInput, allowing optional phone number entry.
- Enhanced form validation schema to include phone number validation.
- Added utility functions for formatting US and Canadian phone numbers.

darekrossman committed:
refactor: simplify search component

ah100101 commented on lib/sfcc/reshape.ts:
```diff
@@ -94,16 +88,12 @@ export function reshapeProduct(product: ShopperProductsTypes.Product) {
-      },
+      // In the unlikely case of a missing price, we set a default of $9999.99 to avoid listing items
+      // as $0. This is preferrable for most merchants to avoid having to honor free purchases.
+      max: product.priceMax?.toString() || product.price?.toString() || "9999.99",
+      min: product.price?.toString() || "9999.99",
```
If possible I would like to avoid changing the shape of the Product that is returned to remain mostly inline with the original product schema, but understand the schema is fairly different and we reference `max` and `min` in newly updated components (e.g. `product/[handle]/page.tsx`)

Perhaps we keep `maxVariantPrice` & `minVariantPrice` as what's returned but set to the `product.priceMax` and `product.price` as you have it? What do you think?

```
maxVariantPrice: {
  amount: product.priceMax?.toString() || product.price?.toString() || "9999.99",
  currencyCode: product.currency || "USD",
},
minVariantPrice: {
  amount: product.price?.toString() || "9999.99",
  currencyCode: product.currency || "USD",
},
```

darekrossman replied:
hard agree, will do

ah100101 commented on lib/sfcc/types.ts:
```diff
@@ -21,12 +21,14 @@ export type SalesforceProduct = {
   priceRange: {
-    maxVariantPrice: Money;
-    minVariantPrice: Money;
+    max: string;
+    min: string;
```
See other comment on changing shape of product.

ah100101 reviewed (changes requested):
Hey @darekrossman - Changes look mostly good, glad that we're adding support for ranges in SFCC. I have one item of feedback and possible change around changing the shape of product.

darekrossman committed:
refactor: update product pricing structure across components

- Modified price handling in ProductPage, Carousel, and other components to use minVariantPrice and maxVariantPrice for better accuracy.
- Adjusted reshapeProduct function to align with new pricing structure.
- Updated types to reflect changes in priceRange properties.

darekrossman committed:
refactor: simplify CartModal item display

- Replaced the conditional rendering of selected options with a simpler check for the merchandise title.
- Updated the display logic to show the merchandise title when it differs from the default option, enhancing clarity and reducing complexity.
